src/drivers/can.rs

```diff
@@ -0,0 +1,1502 @@
+            RXBN::RXB1 => InstructionCommand::ReadRxb1 as u8,
+        };
+
+        /* Start of the transaction. */
+        self.cs.set_low();
+        let mut instruction_buf = [instruction];
+        self.spi.write(&[instruction])?;
+        self.spi.transfer(&mut rx_buffer, &[0; 13]).ok(); // Read the 14 bytes of CAN data
+        self.cs.set_high();
+        /* End of transation. */
+
+        let mut frame = CanMessage::try_from(rx_buffer);
+        //defmt::println!("Reading RX buffer (received): {:08b}", rx_buffer);
+
+        Ok(rx_buffer)
+    }
+
+    /// Reset the MCP2515 and its registers to its default values.
+    /// This is usually called, during startup, before configuration.
+    fn reset_instruction(&mut self) {
+        defmt::println!("Calling Reset instruction over SPI");
+        self.cs.set_low();
+        self.spi.write(&[InstructionCommand::Reset as u8]);
+        self.cs.set_high();
+    }
+
+    /// This would CLEAR the TXREQ bits indicating the tx buffer is not pending
+    /// before writing. When `is_pending` flag is false, it indicate that
+    /// transmit buffer is not pending transmission. Hence should be called
+    /// before writing to the transmit buffer. Setting the flag to true,
+    /// would flag a message buffer as being ready for transmission.
+    ///
+    /// ## TODO
+    ///
+    /// Add which TX buffer to set as ready or pending, by adding argument.
+    /// This is important so we dont CLEAR or set something that is not ready.
+    fn tx_pending(&mut self, is_pending: bool) -> &mut Self {
+        //defmt::info!("Logic for tx_pending:");
+        const CLEAR: u8 = 0u8;
+        const FILTERMASKOFF: u8 = 0b0110_0000;
+        let mut reg_bits = 0u8;
+        reg_bits |= 1 << 3;
+        let bit_mask: u8 = 0b0000_1000;
+        //reg_bits |= 0b0000_0011;
+
+        if (is_pending) {
+            //defmt::println!("is_pending transmission = true\nWriting {:08b} to TXB0CTRL",
+            // reg_bits);
+            self.write_register(MCP2515Register::TXB0CTRL, bit_mask);
+            //self.write_register(MCP2515Register::TXB1CTRL, reg_bits);
+            //self.write_register(MCP2515Register::TXB2CTRL, reg_bits);
+        } else {
+            self.write_register(MCP2515Register::TXB0CTRL, CLEAR);
+            //self.write_register(MCP2515Register::TXB1CTRL, CLEAR);
+            //self.write_register(MCP2515Register::TXB2CTRL, CLEAR);
+            self.write_register(MCP2515Register::RXB0CTRL, CLEAR);
+            self.write_register(MCP2515Register::RXB1CTRL, CLEAR);
+
+            //self.write_register(MCP2515Register::RXB0CTRL, FILTERMASKOFF);
+            //self.write_register(MCP2515Register::RXB1CTRL, FILTERMASKOFF);
+        }
+        self
+    }
+
+    /// This will set the RXBNCTRL.RXM bits, for setting the RX Operating Mode.
+    ///
+    /// ## NOTE
+    ///
+    /// RXM: Receive Buffer Operating Mode bits
+    ///         ################################################
```

Comment:
Look at cargo doc --open to see how this looks.

Reply:
![Screenshot 2024-10-24 at 18-08-11 controller drivers can - Rust](https://github.com/user-attachments/assets/457ed05d-4f20-4312-aff6-85e3a9c6ed2c)
cargo doc seems to work looks like this atm. Still need some fine-tuning!